src/web3-contracts.js

```diff
@@ -20,3 +22,19 @@ export function useContract(address, abi, signer = true) {
     )
   }, [abi, account, address, ethers, signer])
 }
+
+export function useContractReadOnly(address, abi) {
+  const ethEndpoint = networkConfigs[getNetworkType()].nodes.defaultEth
+
+  const ethProvider = useMemo(
+    () => (ethEndpoint ? new Providers.JsonRpcProvider(ethEndpoint) : null),
```

Comment:
A note for later: I think for now this is totally fine, since this provider uses http, but when we switch to websockets, we'll most likely want to initialize a singleton provider to maintain a single websocket connection for all contracts.

Reply:
ok, let's keep that in mind